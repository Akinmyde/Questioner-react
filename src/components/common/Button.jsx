import React from 'react';
import propTypes from 'prop-types';

const Button = ({ value, ...rest }) => (<button {...rest} type="button">{value}</button>);

Button.defaultProps = {
  className: propTypes.defaultProps = 'btn',
};

Button.propTypes = {
  id: propTypes.string.isRequired,
  value: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
  className: propTypes.string,
};

export default Button;
