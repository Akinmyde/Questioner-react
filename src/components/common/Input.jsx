/* eslint-disable jsx-a11y/no-autofocus */
/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import propTypes from 'prop-types';

const Input = ({ id, label, ...rest }) => (
  <div className="form-group">
    <label htmlFor={id}>{label}</label>
    <input
      {...rest}
      id={id}
    />
  </div>
);

Input.defaultProps = {
  autoFocus: propTypes.defaultProps = '',
};

Input.propTypes = {
  type: propTypes.string.isRequired,
  id: propTypes.string.isRequired,
  required: propTypes.string.isRequired,
  label: propTypes.string.isRequired,
  autoFocus: propTypes.string,
  value: propTypes.string.isRequired,
  placeholder: propTypes.string.isRequired,
  onChange: propTypes.func.isRequired,
};

export default Input;
