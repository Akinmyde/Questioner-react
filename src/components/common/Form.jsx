import React, { Component } from 'react';
import { toast } from 'react-toastify';
import validate from '../../helpers/validator';
import Input from './Input';
import Button from './Button';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = { data: {} };
  }

  handelClick = (e) => {
    e.preventDefault();
    const { data } = this.state;
    const error = validate(data, this.schema);
    if (error) return toast.error(error);
    return this.doSubmit();
  };

  handleChange = ({ currentTarget: input }) => {
    const { data } = this.state;
    const account = { ...data };
    account[input.id] = input.value;
    this.setState({ data: account });
  };

  renderButton = (label, className) => (
    <Button onClick={this.handelClick} className={className} value={label} />
  )

  renderInput = (id, label, placeholder, type = 'text') => {
    const { data } = this.state;
    return (
      <Input
        value={data[id]}
        onChange={this.handleChange}
        type={type}
        id={id}
        label={label}
        placeholder={placeholder}
      />
    );
  }

  render() {
    return null;
  }
}

export default Form;
