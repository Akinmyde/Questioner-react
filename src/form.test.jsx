import 'babel-polyfill';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../src/components/common/Form';
import Input from './components/common/Input';
import Button from './components/common/Button';

describe('HomePage component', () => {
  it('should match snapshot', () => {
    const wrapper = <Form />;
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot renderInput', () => {
    const wrapper = renderer.create(<Form />);
    const inst = wrapper.getInstance();
    expect(inst.renderInput('text', 'text')).toMatchSnapshot();
    expect(inst.renderInput('test1', 'test1', 'password')).toMatchSnapshot();
    expect(inst.renderInput('test2', 'test2', 'email')).toMatchSnapshot();
  });

  it('should match snapshot renderButton', () => {
    const wrapper = renderer.create(<Form />);
    const inst = wrapper.getInstance();
    expect(inst.renderButton('test', 'active')).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const wrapper = shallow(<Form />);
    expect(wrapper.find('div'));
  });
});
