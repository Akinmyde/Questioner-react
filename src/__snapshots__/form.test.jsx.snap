// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should match snapshot 1`] = `<Form />`;

exports[`HomePage component should match snapshot renderButton 1`] = `
<Button
  className="active"
  onClick={[Function]}
  value="test"
/>
`;

exports[`HomePage component should match snapshot renderInput 1`] = `
<Input
  autoFocus=""
  id="text"
  label="text"
  onChange={[Function]}
  type="text"
/>
`;

exports[`HomePage component should match snapshot renderInput 2`] = `
<Input
  autoFocus=""
  id="test1"
  label="test1"
  onChange={[Function]}
  placeholder="password"
  type="text"
/>
`;

exports[`HomePage component should match snapshot renderInput 3`] = `
<Input
  autoFocus=""
  id="test2"
  label="test2"
  onChange={[Function]}
  placeholder="email"
  type="text"
/>
`;
